clang-tidy/action.yaml

```
@@ -0,0 +1,78 @@
+name: clang-tidy
+description: Run clang-tidy on the source files
+
+inputs:
+  rosdistro:
+    description: ROS distro
+    required: true
+  build-depends-repos:
+    description: VCS repositories containing build dependencies
+    required: true
+  target-files:
+    description: Target files to run clang-tidy on
+    required: true
+  compile-commands-hash:
+    description: Calculated hash of compile_commands.json
+    required: true
+  token:
+    description: If the repository is private, specify a token.
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    - name: Install pip
+      run: |
+        sudo apt-get -y -q update
+        sudo apt-get -y -q install python3-pip
+      shell: bash
+
+    - name: Install Clang-Tidy
+      run: |
+        sudo apt-get -y -q update
+        sudo apt-get -y -q install clang-tidy
+      shell: bash
+
+    - name: Show target files
+      run: |
+        echo "target files: ${{ inputs.target-files }}"
+      shell: bash
+
+    - name: Set git config for private repositories
+      run: |
+        git config --local --unset-all http.https://github.com/.extraheader || true
+        git config --global url.https://${{ inputs.token }}@github.com.insteadOf 'https://github.com'
```

I'm sorry, I forgot to modify this before.
Adding `x-access-token:` is more useful, I remember.

```suggestion
        git config --global url.https://x-access-token:${{ inputs.token }}@github.com.insteadOf 'https://github.com'
```